tune_sklearn/tune_search.py

```diff
@@ -176,18 +192,68 @@ def __init__(self,
                  local_dir="~/ray_results",
                  max_iters=10,
                  search_optimization="random",
-                 use_gpu=False):
-
-        if (search_optimization not in ["random", "bayesian"]):
-            raise ValueError("Search optimization must be random or bayesian")
-        if (search_optimization == "bayesian" and random_state is not None):
+                 use_gpu=False,
+                 **kwargs):
+
+        search_optimization = search_optimization.lower()
+        available_optimizations = [
+            "random",
+            "bayesian",  #scikit-optimize/SkOpt
+            "bohb",
+            "hyperopt",
+            #"optuna",  # optuna is not yet in stable ray.tune
+        ]
+        if (search_optimization not in available_optimizations):
+            raise ValueError(
+                f"Search optimization must be one of {', '.join(available_optimizations)}"
+            )
+        if (search_optimization != "random" and random_state is not None):
             warnings.warn(
-                "random state is ignored when using Bayesian optimization")
+                "random state is ignored when not using Random optimization")
+
+        if search_optimization == "bayesian":
+            try:
+                import skopt
+                from skopt import Optimizer
+                from ray.tune.suggest.skopt import SkOptSearch
+            except:
+                traceback.print_exc(file=sys.stderr)
+                print(
+                    "It appears that scikit-optimize is not installed. Do: pip install scikit-optimize",
+                    file=sys.stderr)
+        elif search_optimization == "bohb":
+            try:
+                from ray.tune.suggest.bohb import TuneBOHB
+                from ray.tune.schedulers import HyperBandForBOHB
+                import ConfigSpace as CS
+            except:
+                traceback.print_exc(file=sys.stderr)
+                print(
+                    "It appears that either HpBandSter or ConfigSpace is not installed. Do: pip install hpbandster ConfigSpace",
+                    file=sys.stderr)
+        elif search_optimization == "hyperopt":
+            try:
+                from ray.tune.suggest.hyperopt import HyperOptSearch
+                from hyperopt import hp
+            except:
+                traceback.print_exc(file=sys.stderr)
+                print(
+                    "It appears that hyperopt is not installed. Do: pip install hyperopt",
+                    file=sys.stderr)
+        elif search_optimization == "optuna":
+            try:
+                from ray.tune.suggest.optuna import OptunaSearch, param
+                import optuna
+            except:
+                traceback.print_exc(file=sys.stderr)
+                print(
+                    "It appears that optuna is not installed. Do: pip install optuna",
+                    file=sys.stderr)
```

Comment:
nit: can we move this library-checking validation logic into a `_try_import_library()`? same with the above